### PR TITLE
chore: update recommended node version to 22

### PR DESCRIPTION
**Pull request by Akari0731**
Since node v22 will become LTS in next week, we can safely recommend v22 for this repo. Setting engine configuration is just a recommendation so updating here for documentation purpose.
https://nodejs.org/en/about/previous-releases

### CODE REVIEW

File: dist/hooks/useDebouncedValue/useDebouncedValue.d.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/**
```
Comment (odsod) [inline dist/hooks/useDebouncedValue/useDebouncedValue.d.ts:1]: Did something related to the v22 bump trigger the generation of these new files?

File: .prettierignore [hunk 2]
```diff
@@ -0,0 +1 @@
+dist/
```
Comment (Akari0731) [inline .prettierignore:1]: We don't need to apply formatter in dist folder

File: rollup.config.mjs [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import typescript from "@rollup/plugin-typescript"
-import pkg from "./package.json" assert { type: "json" }
+import pkg from "./package.json" with { type: "json" }
```
Comment (Akari0731) [inline rollup.config.mjs:2]: node 22 doesn't support `assert` here, fixed it by using `with`

File: vite.config.ts [hunk 4]
```diff
@@ -1,5 +1,4 @@
 /// <reference types="vitest" />
-/// <reference types="vite/client" />
```
Comment (Akari0731) [inline vite.config.ts:2]: This is not needed

File: dist/hooks/useDebouncedValue/useDebouncedValue.d.ts (hunk 1 above)
Comment (Akari0731) [reply dist/hooks/useDebouncedValue/useDebouncedValue.d.ts]: No it was my VSCode setting formatted all files, reverted those changes:) @odsod

**Review by odsod — APPROVED** [review]

**Pull request merged**